### PR TITLE
Docs: Update to links from static doc pages

### PR DESCRIPTION
**Pull request by mlange05**
_~Filing early to get the resulting change and errors logs. Please ignore for now...~_

General documentation update following the move of sub-packages. This primarily does three things:
1. Update static links in the documentation pages to sub-packages that have moved
2. General update of docstrings that trigger silent errors or warnings in the docs build
3. Addition of sub-package docstrings to give a brief overview of the respective sub-packages (the auto-generated API list looked empty to me).

Unfortunately, I didn't get everything in the docstring clean-up commit, and there are still a few warnings I do not understand. The overall impact should be net-positive though.

### CODE REVIEW

File: loki/build/__init__.py [hunk 1]
```diff
@@ -4,6 +4,13 @@
 # In applying this licence, ECMWF does not waive the privileges and immunities
 # granted to it by virtue of its status as an intergovernmental organisation
 # nor does it submit to any jurisdiction.
+"""
+Just-in-Time compilation utilities used in the Loki test base.
+
+These allow compilation and wrapping of generated Fortran source code
+using `f90wrap <https://github.com/jameskermode/f90wrap>` for
```
Comment (reuterbal) [inline loki/build/__init__.py:11]:
```suggestion
using `f90wrap <https://github.com/jameskermode/f90wrap>`_ for
```

File: loki/expression/__init__.py [hunk 2]
```diff
@@ -4,6 +4,10 @@
 # In applying this licence, ECMWF does not waive the privileges and immunities
 # granted to it by virtue of its status as an intergovernmental organisation
 # nor does it submit to any jurisdiction.
+"""
+Expression layer of the two-level Loki IR based on `Pymbolic
+<https://github.com/inducer/pymbolic`.
```
Comment (reuterbal) [inline loki/expression/__init__.py:9]:
```suggestion
Expression layer of the two-level Loki IR based on
`Pymbolic <https://github.com/inducer/pymbolic>`_.
```

File: loki/frontend/__init__.py [hunk 3]
```diff
@@ -4,6 +4,13 @@
 # In applying this licence, ECMWF does not waive the privileges and immunities
 # granted to it by virtue of its status as an intergovernmental organisation
 # nor does it submit to any jurisdiction.
+"""
+Frontend parsers that create Loki IR from input Fortran code.
+
+This includes code sanitisation utilities and several frontend parser
+interfaces, including the REGEX-frontend that used for fast source
```
Comment (reuterbal) [inline loki/frontend/__init__.py:11]:
```suggestion
interfaces, including the REGEX-frontend that is used for fast source
```

File: loki/frontend/__init__.py [hunk 4]
```diff
@@ -4,6 +4,13 @@
...
+code exploration in large call trees.
```
Comment (reuterbal) [inline loki/frontend/__init__.py:12]:
```suggestion
code exploration in large call and dependency trees.
```

File: loki/transformations/__init__.py [hunk 5]
```diff
@@ -4,6 +4,13 @@
 # In applying this licence, ECMWF does not waive the privileges and immunities
 # granted to it by virtue of its status as an intergovernmental organisation
 # nor does it submit to any jurisdiction.
+"""
+Sub-package with supported source code transformation passes.
+
+This sub-package includes general source code trnasformations and
```
Comment (reuterbal) [inline loki/transformations/__init__.py:10]:
```suggestion
This sub-package includes general source code transformations and
```

File: loki/batch/__init__.py [hunk 6]
```diff
@@ -4,6 +4,15 @@
 # In applying this licence, ECMWF does not waive the privileges and immunities
 # granted to it by virtue of its status as an intergovernmental organisation
 # nor does it submit to any jurisdiction.
+"""
+Batch processing abstraction for processing large source trees with Loki.
+
+This sub-package provides the :any:`Scheduler` class that allows Loki
+transformations to be applied over large source trees. For this it
+provides the basic :any:`Transformation` and :and:`Pipeline` classes
```
Comment (reuterbal) [inline loki/batch/__init__.py:12]:
```suggestion
provides the basic :any:`Transformation` and :any:`Pipeline` classes
```

**Review by reuterbal — CHANGES_REQUESTED** [review]
Love this, many thanks! I've flagged a few typos, otherwise good to go!

**Comment by mlange05** [comment]
Ok, thanks @reuterbal . Pushed fixes and rebased.

**Review by reuterbal — APPROVED** [review]
Thanks! There was a 1-letter typo still, which I took the liberty of fixing myself. Going in...

**Pull request merged**